src/OnlineRetailPortal.MongoDBStore/MongoProductStore.cs

```diff
@@ -47,10 +47,14 @@ public async Task<GetProductStoreResponse> GetProductAsync(string productId)
             try
             {
                 mongoEntity = await productStoreCollection.Find(x => x.Id == productId).FirstOrDefaultAsync();
+                if (mongoEntity == null)
+                {
+                    throw new BaseException(int.Parse(ErrorCode.ProductNotFound()), Error.ProductNotFound(productId), null, HttpStatusCode.NotFound);
+                }
             }
-            catch
+            catch(BaseException baseException)
```

Comment:
if you want to throw same exception then why are we catching it here ?

Reply:
okay, removed try catch